dev/com.ibm.ws.security.social/src/com/ibm/ws/security/social/internal/Oauth2LoginConfigImpl.java

```diff
@@ -268,6 +272,19 @@ boolean isConfiguredForProxyFlow(Map<String, Object> props) {
     protected void checkForRequiredConfigAttributesForProxyFlow(Map<String, Object> props) {
         configUtils.getRequiredConfigAttributeWithConfigId(props, KEY_userApi, uniqueId);
     }
+    
+    boolean isIntrospectConfiguration(Map<String, Object> props) {
+        String userApiType = configUtils.getConfigAttribute(props, KEY_userApiType);
+        if (userApiType != null && "introspect".equals(userApiType)) {
+            return true;
+        }
+        return false;
+    }
+    
+    protected void checkForRequiredConfigAttributesForIntrospect(Map<String, Object> props) {
+        configUtils.getRequiredConfigAttribute(props, KEY_clientId);
```

Comment:
I'd switch this to use the `getRequiredConfigAttribute()` method that's defined in this class. If you keep this call as you have it now, this will emit a message that says
> CWWKS6104W: The configuration attribute [clientId] that is required is missing or empty and a default value is not provided. Verify that the attribute is configured, that it is not empty, and that it does not consist of only white space characters.

That's technically correct, but the message doesn't tell us _which element_ is missing that attribute. Using the other method will emit this error message:
>CWWKS5479E: The configuration attribute [clientId] that is required in the social login configuration [someSpecificSocialRP] is missing or empty. Verify that the attribute is configured, that it is not empty, and that it does not consist of only white space characters.

That helps a server admin because then they know that the `someSpecificSocialRP` element was the one with the problem. A server could have a bunch of social login configurations, so not knowing which specific one caused the problem would be a pain.